python/taichi/lang/matrix.py

```diff
@@ -411,6 +411,21 @@ def diag(dim, val):
     def loop_range(self):
         return self.entries[0]
 
+    def shape(self):
+        # Took `self.entries[0]` as a representation of this tensor-of-matrices.
+        # https://github.com/taichi-dev/taichi/issues/1069#issuecomment-635712140
+        return self.loop_range().shape()
+
+    def dim(self):
+        return self.loop_range().dim()
+
+    def __iter__(self):
+        return iter(self.loop_range())
+
+    #def data_type(self):
+    #    # XXX(@yuanming-hu): Do we need to tell users we are a matrix of dt?
+    #    return self.loop_range().data_type()
```

Comment:
Or user can easy get confused when:
```py
@ti.kernel
def clear(x: ti.template()):
  x[None] = ti.cast(x.data_type(), 0)

scalar = ti.var(dt=ti.i32, shape=())
matrix = ti.Matrix(2, 3, dt=ti.i32, shape=())

clear(scalar)  # OK
clear(matrix)  # Error: can not cast from 0 (int) to x[None] (Matrix of int)
``
Thoughts? @vkurenkov @yuanming-hu @k-ye

Reply:
Good point! I think we should add `data_type` to `Matrix`.